### ISSUE TITLE
Prevent failure on plugin self-removal with post hooks hooked in

### ISSUE THREAD

**Issue by simPod** [composer/composer#8691]
Currently, each plugin has to handle its own removal. Typical example with `vendor/lib` that adds post install/update hooks into composer:

```
composer require --dev vendor/lib
composer install --no-dev
```

After executing `composer install --no-dev` the hooks from `vendor/lib` fail before the lib is removed and its source files are not present anymore, though post hooks are still active.

Shouldn't it be rather resolved here? From my experience most of plugins forget to implement this self-removal check. This seems like followup for https://github.com/composer/composer/issues/3729.

See:

https://github.com/phpstan/extension-installer/issues/3  
https://github.com/malukenho/mcbumpface/pull/30  
https://github.com/Roave/you-are-using-it-wrong/pull/63

**Comment by Seldaek** [comment]
This is fixed in 2.0 branch already as it offers a new interface for this: https://github.com/composer/composer/blob/2.0/src/Composer/Plugin/PluginInterface.php#L50

**Comment by simPod** [comment]
Thank you for reply!
